db/schema_tables.cc

```diff
@@ -299,19 +299,22 @@ schema_ptr tables() {
 }
 
 // Holds Scylla-specific table metadata.
-schema_ptr scylla_tables() {
-    static thread_local auto schema = [] {
+schema_ptr scylla_tables(schema_features features) {
```

Comment:
I think that instead of passing `features` here and making the addition of the `cdc` column conditional on it, we should just remove this column when we need to redact it, during upgrade.
When writing code for mixed cluster support, we want to disturb the common path of a homogeneous cluster the least possible, and do all the support related extra work in the mixed cluster case. Here you add extra code and logic for the vastly more common case of a non-mixed cluster.

Reply:
The schema_features parameter makes it impossible to use the wrong schema accidentally.

If you're worried about performance, this isn't in any hot path, and in any case translates to an array lookup.